src/components/App/App.js

```diff
@@ -1,34 +1,54 @@
-import './App.css';
-import movieData from '../../movieData.js';
-import {useState} from 'react'
-import AllMovies from '../AllMovies/AllMovies.js'
-import MovieDetail from '../MovieDetails/MovieDetail';
+import "./App.css";
+import movieData from "../../movieData.js";
+import { useState, useEffect } from "react";
+import AllMovies from "../AllMovies/AllMovies.js";
+import MovieDetail from "../MovieDetails/MovieDetail";
 
 const App = () => {
-  const [movies, setMovies] = useState(movieData)
-  const [allMoviesView, setAllMoviesView] = useState(true)
-  const [singleMovieView, setSingleMovieView] = useState(false)
-  const [selectedMovieDetails, setSelectedMovieDetails] = useState(null)
-  
+  const [movies, setMovies] = useState(movieData);
+  const [allMoviesView, setAllMoviesView] = useState(true);
+  const [singleMovieView, setSingleMovieView] = useState(false);
+  const [error, setError] = useState(false);
+
+  const getAllMovieData = () => {
+    fetch("https://rancid-tomatillos.herokuapp.com/api/v2/movies")
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error("Server returned " + response.status);
+        }
+        return response.json();
+      })
+      .then((data) => setMovies(data))
+      .catch((error) => {
+        setError(true);
+      });
+  };
+  useEffect(() => {
+    getAllMovieData();
+  }, []);
+
+  const [selectedMovieId, setSelectedMovieId] = useState(null);
```

Comment:
I'd recommend keep all your pieces of state together at the top of the file so other devs can see at a glance what pieces of state are being tracked.  
Also, this is probably more of a dev-opinion rather than true best-practice but consider how it might be beneficial to use an empty state of the expected data type rather than using null here. Personally, I appreciate an empty state because then its clear to me and other devs right off the bat that the selectedSingleMovie state will be an array, object, etc. With null, I have no idea what shape that data will eventually take.